tensorflow_datasets/core/dataset_info.py

```diff
@@ -185,68 +196,69 @@ def full_name(self):
     return posixpath.join(*names)
 
   @property
-  def description(self):
+  def description(self) -> str:
     return self.as_proto.description
 
   @property
-  def version(self):
+  def version(self) -> utils.Version:
     return self._builder.version
 
   @property
-  def homepage(self):
+  def homepage(self) -> str:
     urls = self.as_proto.location.urls
     tfds_homepage = "https://www.tensorflow.org/datasets/catalog/{}".format(
         self.name)
     return urls and urls[0] or tfds_homepage
 
   @property
-  def citation(self):
+  def citation(self) -> str:
     return self.as_proto.citation
 
   @property
-  def data_dir(self):
+  def data_dir(self) -> str:
     return self._builder.data_dir
 
   @property
-  def dataset_size(self):
+  def dataset_size(self) -> int:
     """Generated dataset files size, in bytes."""
     # For old datasets, maybe empty.
     return sum(split.num_bytes for split in self.splits.values())
 
   @property
-  def download_size(self):
+  def download_size(self) -> int:
     """Downloaded files size, in bytes."""
     # Fallback to deprecated `size_in_bytes` if `download_size` is empty.
     return self.as_proto.download_size or self.as_proto.size_in_bytes
 
   @download_size.setter
-  def download_size(self, size):
+  def download_size(self, size: int) -> None:
     self.as_proto.download_size = size
 
   @property
-  def features(self):
+  def features(self) -> features_lib.FeaturesDict:
     return self._features
 
   @property
-  def metadata(self):
+  def metadata(self) -> Metadata:
```

Comment:
As metadata can be optional, so can be the returned value `Optional[Metadata]`

Reply:
Done